sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -174,4 +174,36 @@ class CollationSuite extends QueryTest with SharedSparkSession {
           Row(expected))
     }
   }
+
+
+  test("disable partition on collated string column") {
+    def createTable(partitionColumns: String*): Unit = {
+      val tableName = "test_partition_tbl"
+      withTable(tableName) {
+        sql(
+          s"""
+             |CREATE TABLE $tableName
+             |(id INT, c1 STRING COLLATE 'UNICODE', c2 string)
+             |USING parquet
+             |PARTITIONED BY (${partitionColumns.mkString(",")})
+             |""".stripMargin)
+      }
+
+      // should work fine on non collated columns
+      createTable("id")
+      createTable("c2")
+      createTable("id", "c2")
+
+      Seq(Seq("c1"), Seq("c1", "id"), Seq("c1", "c2")).foreach { partitionColumns =>
+        checkError(
+          exception = intercept[AnalysisException] {
+            createTable(partitionColumns: _*)
+          },
+          errorClass = "_LEGACY_ERROR_TEMP_1153",
```

Comment:
As you are here, could you assign proper name for the error class and improve error message format (if it is needed).

Reply:
sure